src/System.CommandLine.Tests/CommandHandlerTests.cs

```diff
@@ -142,23 +143,42 @@ void Execute(string name, int Age)
         }
 
         [Fact]
-        public async Task Method_parameters_on_the_invoked_lambda_are_bound_to_matching_option_names()
+        public void Method_parameters_on_the_invoked_lambda_are_bound_to_matching_option_names()
         {
             var wasCalled = false;
+            const string commandLine = "command --age 425 --name Gandalf";
 
             var command = new Command("command");
             command.AddOption(new Option("--name", "", new Argument<string>()));
             command.AddOption(new Option("--age", "", new Argument<int>()));
-            command.Handler = CommandHandler.Create<string, int>((name, age) =>
+            var handler = CommandHandler.Create<string, int>((name, age) =>
             {
-                wasCalled = true;
-                name.Should().Be("Gandalf");
-                age.Should().Be(425);
-            });
-
-            await command.InvokeAsync("command --age 425 --name Gandalf", _console);
+                //wasCalled = true;
```

Comment:
This change means this test is testing something fairly different than it was before. I think the old test is still valid, as it's testing invocation with binding along the way. We still need invocation tests even if the new binding approach makes binding-only tests more viable.

Let's roll back this change and put your version in a new test class focused on binding.

Reply:
I copied these tests and have them in a format that tests more completely via Reflection Binding